server/src/main/java/org/elasticsearch/rest/action/cat/RestNodesAction.java

```diff
@@ -92,10 +87,9 @@ protected void documentation(StringBuilder sb) {
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
         clusterStateRequest.clear().nodes(true);
-        if (request.hasParam("local")) {
-            deprecationLogger.deprecated(LOCAL_DEPRECATED_MESSAGE);
+        if (request.hasParam("local") && Version.CURRENT.major == Version.V_7_0_0.major + 1) { // only needed in v8 to catch breaking usages
```

Comment:
Why not `Version.V_8_0_0.major`? :)

Reply:
So we get a compile error when we move to v9 and drop the v7 version constants.